release.sh

```diff
@@ -3,7 +3,7 @@
 RELEASE_BRANCH=master
 RELEASE_DIR=HCK_CI
 AUTO_HCK=AutoHCK
-
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Comment:
the command gets the script's source file pathname, strips it to just the path portion, cds to that path, then uses pwd to return the (effectively) full path of the script. This is assigned to DIR. After all of that, the context is unwound so you end up back in the directory you started at but with an environment variable DIR containing the script's path.

to save the current dir will not work in case you are not running the script within the directory.
Also according to the explaination above the command has no side effects.